### ISSUE TITLE
U2F Security Key not supported in Windows 11

### ISSUE THREAD

**Issue by SystemOfTheCSGO** [flipperdevices/flipperzero-firmware#2877]
### Describe the enhancement you're suggesting.

The current module of security key U2F is not supported by windows.

Reproduction

Use the firmware as usual, get a clean install of windows 11, try to link the security key from flipper to windows, you will get a error: this device is not certified.

### Anything else?

This can be bypassed emulating headers of a certified key. (I’ve seen it done before, at least a homemade microsoft certified’ seckey.)

**Comment by doomwastaken** [comment]
Hello, got a few questions about the build that you are using, is that corporate build or any beta version by any chance?

**Comment by kamilskrzypek4** [comment]
The key for MS products like Windows, O365 needs to be FIDO2, as far as I know flipper is FIDO1. Any chance for FIDO2 on flipper hardware?

**Comment by zhiyan114** [comment]
> The key for MS products like Windows, O365 needs to be FIDO2, as far as I know flipper is FIDO1. Any chance for FIDO2 on flipper hardware?

I feel like it would be a security risk, especially since FIDO2 is usually a passwordless login. Definitely safer to get a separate security key for it.

**Comment by skotopes** [comment]
We'll try to pack it in future updates.

**Comment by SolarSciencePup** [comment]
> We'll try to pack it in future updates.

Any news?

**Comment by skotopes** [comment]
Not really, plans are to complete NFC refactoring, then BadUSB/BadUART, and then U2F/FIDO.

**Comment by Mechcondrid** [comment]
Any news? this would help me out since now i need this for U2F on work repos

**Comment by Kevin1904** [comment]
i am also interested in this

**Comment by dol4inart** [comment]
Hey there. I can help with developing U2F/FIDO :)
But I don't know how work U2F in Windows :(

**Comment by alexh3791** [comment]
I just wanted to chime in and say that I'm really looking forward to FIDO2 on the Flipper as well. 

Adding NFC support to the U2F/FIDO app would be extremely helpful, too, since I authenticate from an iPhone more often than a laptop/desktop at this point.

**Comment by randoassuser** [comment]
Does the Flipper (even with XFW) really not support FIDO2? It does not work on my iPhone via lightning cable I‘m so disappointed like tf

**Comment by bomkz** [comment]
> Not really, plans are to complete NFC refactoring, then BadUSB/BadUART, and then U2F/FIDO.

From what I see searching around the repo, NFC refactoring has been completed, I may be wrong and/or misinterpreted, I'm not sure about whether BadUSB/BadUART has been, but if so, is FIDO2 support still in the pipeline?

**Comment by skotopes** [comment]
In a long term FIDO2 is on the roadmap. But right now no one is working on it and we also'd like to have proper secure enclave before going further.

**Comment by exussum12** [comment]
https://github.com/polhenarejos/pico-fido

That is compatible with an ESP32 Not sure what would be needed to get STM32 support  but posibly part way there.

This spoofs VID too gives better compatibility

**Comment by Luk164** [comment]
Has there been any news on FIDO2 support lately?

**Comment by Daniel53245** [comment]
Hi, is there still  any plan on futher support FIDO2 on flipper  official firmware ? I believe there need extra chip for security so it might not be packed in the official firmware ?

**Comment by Sporksan** [comment]
Without getting certified in the FIDO MDS many commercial Identity Providers will not work with a Flipper running FIDO2, unless they copy the AAGUID of a known Hardware Key.  This means that systems like Microsoft and GitHub would likely reject the Flipper as a FIDO2 device.  Flipper *Could* clone a AAGUID, but it would not have access to the root certs to properly sign things and any authenticator validation would likely fail.

**Comment by exussum12** [comment]
the pico-fido project works well, if flipper used that everything would work

**Comment by Luk164** [comment]
@exussum12 Would it be viable to copy their approach? AFAIK ESP32-S2/3 does have an actual secure element, which is what allows it to do this. Maybe a hardware mod adding some sort of I2C TPM could work?

**Comment by exussum12** [comment]
Would prefer someone on the project to look, Looking at it, I think its possible but not easy :(
